packages/starknet-snap/src/getErc20TokenBalance.ts

```diff
@@ -53,23 +49,43 @@ export async function getErc20TokenBalance(params: ApiParams) {
       `getErc20Balance:\nerc20Address: ${erc20Address}\nuserAddress: ${userAddress}`,
     );
 
-    // For deployed accounts, use the PENDING block to show balance updates as soon as possible,
-    // facilitating concurrent transactions without delays.
-    // For non-deployed accounts, use the LATEST block to avoid displaying non-zero balances
-    // from pending transactions, because confirmed non-zero balance is required for deployment.
-    const blockIdentifier = (await isAccountDeployed(network, userAddress))
-      ? BlockIdentifierEnum.Pending
-      : BlockIdentifierEnum.Latest;
-    const balance = await getBalance(
-      userAddress,
-      erc20Address,
-      network,
-      blockIdentifier,
+    // Always get both balance from pending and latest
+    // Always show the smallest balance of the two.
+    // If user is getting token it can't spend them until they are latest
+    // If user is sending token it can't spend them even if they are on pending.
+    const balanceLatest = BigInt(
+      await getBalance(
+        userAddress,
+        erc20Address,
+        network,
+        BlockIdentifierEnum.Latest,
+      ),
+    );
+    const balancePending = BigInt(
+      await getBalance(
+        userAddress,
+        erc20Address,
+        network,
+        BlockIdentifierEnum.Pending,
+      ),
     );
 
-    logger.log(`getErc20Balance:\nresp: ${toJson(balance)}`);
+    // Use cases :
+    // - X token balance, initiate a transaction of 0.1X
+    // ==> pending balance 0.9 X, Latest balance X, spendable is 0.9 X
+    // - X token balance, receives 0.1X
+    // ==> pending balance 1.1 X, Latest balance X, spendable is X
+    const balanceBigInt =
+      balancePending < balanceLatest ? balancePending : balanceLatest;
+    const spendableBalance = `0x${balanceBigInt.toString(16)}`;
+    const totalBalance = `0x${balancePending.toString(16)}`;
+    const resp = {
+      spendableBalance,
```

Comment:
i just think, if this api is for our dapp, why not just return the balancePending and balanceLatest
and let the FE to determined what balance they need to show?

as backend (snap), it may be better to give 2 informations and let the caller to choose their preference?

Reply:
and btw , i think it is better to return non hex value ? (this api was not created by me, but i think most of the api is returning an string number "123123213123124124" rather than "0x123123")
or your wanna do it in other PR?

Reply:
do you mean to return the integer version not the hex version, e.g. instead of '0xa' we return '10' ?

Reply:
yea, it seem more readable

Reply:
but it is up to us at this moment

Reply:
I agree, let's match MM standard. But let's do this in separate PR if ok with you.